decidim-core/app/models/decidim/user_base_entity.rb

```diff
@@ -32,12 +32,50 @@ class UserBaseEntity < ApplicationRecord
     scope :confirmed, -> { where.not(confirmed_at: nil) }
     scope :not_confirmed, -> { where(confirmed_at: nil) }
 
+    # Visible user entities are those that should appear publicly on the
+    # platform, such as on their personal profile page, the GraphQL API or
+    # other places where the user may appear.
+    scope :visible, lambda {
+      profile_published.not_blocked.merge(
+        Decidim::User.tos_accepted.or(
+          Decidim::UserBaseEntity.where.not(type: "Decidim::User")
+        )
+      )
+    }
+
+    # User entities (user groups) that have their profile visible on the
+    # platform from the user's own perspective. This includes also blocked users
+    # because the decision to hide their profile has been made by someone else.
+    scope :profile_published, -> { confirmed.where(deleted_at: nil, managed: false) }
```

Comment:
Maybe would better to apply scopes here, while we move managed / not_managed scopes to parent class? like 
```suggestion
    scope :profile_published, -> { confirmed.not_deleted.not_managed }
```